docs/prow/how-to_use_tekton-pipelines-in-prow.md

```diff
@@ -0,0 +1,233 @@
+# Use Tekton Pipelines with Prow
+
+Kyma Prow instance supports defining and using Tekton Pipelines as a workload. This gives the developer the ability to use
+Kubernetes-native implementation of CI/CD pipelines, where every task is a self-contained set of commands.
+
+**DISCLAIMER: This document is not complete documentation of Tekton's functionality. This guide is meant only to let you start using Tekton with Prow. 
+For complete information about Tekton's features, read [Tekton official documentation](https://tekton.dev/docs/).**
+
+Tekton can't trigger pipelines natively without an external hook. Prow acts as such a trigger and executes pipeline definition
+when it receives a webhook event from GitHub, just as how ProwJob execution works.
+
+## Defining a PipelineRun in ProwJob
+
+ProwJob's API defines a [`TektonPipelineRunSpec`](https://github.com/kubernetes/test-infra/blob/24e9f4faa85b8504dc8d30b11534a21547159c1e/prow/apis/prowjobs/v1/types.go#L201) specification
+which is compatible with `v1beta1` API version of Tekton PipelineRunSpec.
+
+> **NOTE:** API `v1beta1` is a relatively new addition to Prow. Some bugs and missing features are expected! Raise an issue with the `kind/bug` label if some functionality is not working.
+
+To get further understanding, see the defined simple presubmit ProwJob that uses Tekton Pipelines as an agent:
+```yaml
+presubmits:
+  kyma-project/kyma:
+    - name: pull-ci-tekton-run
+      decorate: false
+      agent: tekton-pipeline
+      cluster: tekton-pipelines
+      always_run: true
+      tekton_pipeline_run_spec:
+        v1beta1:
+            pipelineRef:
+              name: prototype-pipeline
```

Comment:
Using here a pipelineRef, in the next section providing explanation for pipelineSpec and providing explanation for pipelineRef at the end of documentation might confuse users. I would use here a pipelineSpec and mention and explain usage of pipelineRef at the end.

Reply:
The idea is to show difference between Kubernetes agent and Tekton pipeline definition in ProwJob, at this level it's not important for developer to know the inner requirements of a pipeline ProwJob.

Reply:
I know the idea but it still can confuse reader.

Reply:
Then what would be your suggestion to that?

Reply:
```suggestion
            pipelineSpec:
              description: "Hello world!"
                tasks:
                  - name: hello
                     taskSpec:
                       steps:
                         - name: echo
                            image: alpine:edge
                            script: |
                              #!/bin/sh
                              echo "Hello World"
```